smoke/src/main/java/com/adobe/cq/cloud/testing/it/smoke/rules/ContentPublishRule.java

```diff
@@ -0,0 +1,334 @@
+
+                    if (pkgJson != null) {
+                        Set<String> paths = elementsAsText(pkgJson.get("paths"));
+
+                        if (paths.contains(replicatedPath)) {
+                            log.warn("Package [{}] in queue [{}] with paths {} failed due to [{}]", pkg, queueId,
+                                paths, pkgJson.get("errorMessage"));
+                        }
+                    }
+                    return false;
+                }
+            }
+        }
+        return true;
+    }
+    
+    private static Set<String> elementsAsText(JsonNode queue) {
+        return elements(queue).map(JsonNode::getTextValue).collect(Collectors.toSet());
+    }
+
+    private static Stream<JsonNode> elements(JsonNode node) {
+        if (node == null ) {
+            return Stream.empty();
+        }
+        Iterator<JsonNode> elementsIt = node.getElements();
+        return StreamSupport.stream(Spliterators.spliteratorUnknownSize(elementsIt, Spliterator.ORDERED), false);
+    }
+
+
+    private void checkPage(final int... expectedStatus) throws Exception {
+        checkPage(true, expectedStatus);
+    }
+    
+    /**
+     * Checks that a GET on the page on publish has the {{expectedStatus}} in the response
+     *
+     * @throws Exception if an error occurred
+     */
+    private void checkPage(boolean skipDispatcherCache, final int...  expectedStatus) throws Exception {
+        final String path = root.getPath() + ".html";
+        log.info("Checking page {} returns status {}", getPublishClient().getUrl(path), expectedStatus);
+        SlingHttpResponse res = null;
+        final List<NameValuePair> queryParams = skipDispatcherCache
+            ? Collections.singletonList(
+            new BasicNameValuePair("timestamp", String.valueOf(System.currentTimeMillis())))
+            : Collections.emptyList();
+
+        res = getPublishClient().doGet(path, queryParams, Collections.emptyList());
+        if (null != res && res.getStatusLine().getStatusCode() == SC_UNAUTHORIZED) {
+            throw new AssumptionViolatedException("Publish requires auth for (SAML?). Skipping...");
+        }
+
+        try {
+            new Polling() {
+                @Override public Boolean call() throws Exception {
+                    final List<NameValuePair> queryParams = skipDispatcherCache ?
+                        Collections.singletonList(
+                            new BasicNameValuePair("timestamp", String.valueOf(System.currentTimeMillis()))) :
+                        Collections.emptyList();
+                    getPublishClient().doGet(path, queryParams, Collections.emptyList(), expectedStatus);
+                    return true;
+                }
+            }.poll(TIMEOUT_PER_TRY, 1000);
+        } catch (TimeoutException te) {
+            log.warn("Checking page {} with expected status {} failed. Please check that the connectivity to {} is proper",
```

Comment:
Ack. Pre existing log message will change accordingly.

Reply:
Updated, the logs after change will be as below. Since, the call `Poll` only returns a boolean we don't have access to the returned status. Will need a change in the polling method to add a custom exception and/or message.

`[main] INFO com.adobe.cq.cloud.testing.it.smoke.rules.ContentPublishRule - Checking page https://host1/content/test-site/testpage_6b132787-1d52-49b1-8af8-8a96948faf62.html returns status [404]
[main] WARN com.adobe.cq.cloud.testing.it.smoke.rules.ContentPublishRule - Failed to check the page /content/test-site/testpage_6b132787-1d52-49b1-8af8-8a96948faf62 via the AEM publish ingress (expected status [404]). Please ensure that the CDN and Dispatcher configurations allow fetching the page.
`